### PR TITLE
Apple Silicon reenable 49110 failures

### LINKED ISSUE #49110: [macOS-arm64] XUnit crashes while running libraries tests on CoreCLR
See https://dev.azure.com/dnceng/public/_build/results?buildId=1021947

```console
----- start Wed Mar 3 17:40:06 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Xml.Xsl.XslCompiledTransformApi.Tests.runtimeconfig.json --depsfile System.Xml.Xsl.XslCompiledTransformApi.Tests.deps.json xunit.console.dll System.Xml.Xsl.XslCompiledTransformApi.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/9A3508C1/e /private/tmp/helix/working/C24F0A5F/w/9A3508C1/e
  Discovering: System.Xml.Xsl.XslCompiledTransformApi.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Xml.Xsl.XslCompiledTransformApi.Tests (found 282 of 285 test cases)
  Starting:    System.Xml.Xsl.XslCompiledTransformApi.Tests (parallel test collections = on, max threads = 8)
./RunTests.sh: line 162: 28346 Bus error: 10           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Xml.Xsl.XslCompiledTransformApi.Tests.runtimeconfig.json --depsfile System.Xml.Xsl.XslCompiledTransformApi.Tests.deps.json xunit.console.dll System.Xml.Xsl.XslCompiledTransformApi.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/9A3508C1/e
----- end Wed Mar 3 17:40:09 PST 2021 ----- exit code 138 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:39:24 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Xml.Misc.Tests.runtimeconfig.json --depsfile System.Xml.Misc.Tests.deps.json xunit.console.dll System.Xml.Misc.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/A9ED0917/e /private/tmp/helix/working/C24F0A5F/w/A9ED0917/e
  Discovering: System.Xml.Misc.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Xml.Misc.Tests (found 3 test cases)
  Starting:    System.Xml.Misc.Tests (parallel test collections = on, max threads = 8)
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 286
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 277
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 302
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 328
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 299
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs:line 708
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:line 57
   at System.Threading.Thread.StartCallback() in /_/src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 105
./RunTests.sh: line 162: 23723 Abort trap: 6           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Xml.Misc.Tests.runtimeconfig.json --depsfile System.Xml.Misc.Tests.deps.json xunit.console.dll System.Xml.Misc.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/A9ED0917/e
----- end Wed Mar 3 17:39:25 PST 2021 ----- exit code 134 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:38:39 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Threading.Tasks.Parallel.Tests.runtimeconfig.json --depsfile System.Threading.Tasks.Parallel.Tests.deps.json xunit.console.dll System.Threading.Tasks.Parallel.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/9D1008F2/e /private/tmp/helix/working/C24F0A5F/w/9D1008F2/e
  Discovering: System.Threading.Tasks.Parallel.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Threading.Tasks.Parallel.Tests (found 177 test cases)
  Starting:    System.Threading.Tasks.Parallel.Tests (parallel test collections = on, max threads = 8)
Unhandled exception. System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'state')
   at System.Threading.Tasks.Tests.ParallelForEachAsyncTests.<Cancellation_CancelsIterationAndReturnsCanceledTask_Async>g__InfiniteAsync|16_0()+MoveNext() in /_/src/libraries/System.Threading.Tasks.Parallel/tests/ParallelForEachAsyncTests.cs:line 496
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 303
--- End of stack trace from previous location ---
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 336
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs:line 708
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:line 57
   at System.Threading.Thread.StartCallback() in /_/src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 105
./RunTests.sh: line 162: 37188 Abort trap: 6           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Threading.Tasks.Parallel.Tests.runtimeconfig.json --depsfile System.Threading.Tasks.Parallel.Tests.deps.json xunit.console.dll System.Threading.Tasks.Parallel.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/9D1008F2/e
----- end Wed Mar 3 17:38:49 PST 2021 ----- exit code 134 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:38:18 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Text.RegularExpressions.Tests.runtimeconfig.json --depsfile System.Text.RegularExpressions.Tests.deps.json xunit.console.dll System.Text.RegularExpressions.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/99E10886/e /private/tmp/helix/working/C24F0A5F/w/99E10886/e
  Discovering: System.Text.RegularExpressions.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.RegularExpressions.Tests (found 216 of 217 test cases)
  Starting:    System.Text.RegularExpressions.Tests (parallel test collections = on, max threads = 8)
    System.Text.RegularExpressions.Tests.RegexMatchTests.Match_Timeout_Repetition_Throws [SKIP]
      Condition(s) not met: "IsWindows"
./RunTests.sh: line 162: 27269 Bus error: 10           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.RegularExpressions.Tests.runtimeconfig.json --depsfile System.Text.RegularExpressions.Tests.deps.json xunit.console.dll System.Text.RegularExpressions.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/99E10886/e
----- end Wed Mar 3 17:38:21 PST 2021 ----- exit code 138 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:37:25 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Runtime.Tests.runtimeconfig.json --depsfile System.Runtime.Tests.deps.json xunit.console.dll System.Runtime.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/B4830956/e /private/tmp/helix/working/C24F0A5F/w/B4830956/e
  Discovering: System.Runtime.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Runtime.Tests (found 6066 of 6088 test cases)
  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 8)
./RunTests.sh: line 162: 28347 Bus error: 10           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Runtime.Tests.runtimeconfig.json --depsfile System.Runtime.Tests.deps.json xunit.console.dll System.Runtime.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/B4830956/e
----- end Wed Mar 3 17:37:28 PST 2021 ----- exit code 138 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:36:49 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Private.Uri.Unit.Tests.runtimeconfig.json --depsfile System.Private.Uri.Unit.Tests.deps.json xunit.console.dll System.Private.Uri.Unit.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/B04509BC/e /private/tmp/helix/working/C24F0A5F/w/B04509BC/e
  Discovering: System.Private.Uri.Unit.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Private.Uri.Unit.Tests (found 18 test cases)
  Starting:    System.Private.Uri.Unit.Tests (parallel test collections = on, max threads = 8)
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 286
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 277
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 302
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 328
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecuteFromThreadPool(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 299
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs:line 708
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:line 57
   at System.Threading.Thread.StartCallback() in /_/src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 105
./RunTests.sh: line 162: 26517 Abort trap: 6           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Private.Uri.Unit.Tests.runtimeconfig.json --depsfile System.Private.Uri.Unit.Tests.deps.json xunit.console.dll System.Private.Uri.Unit.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/B04509BC/e
----- end Wed Mar 3 17:36:56 PST 2021 ----- exit code 134 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:36:43 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Numerics.Vectors.Tests.runtimeconfig.json --depsfile System.Numerics.Vectors.Tests.deps.json xunit.console.dll System.Numerics.Vectors.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/B0D20981/e /private/tmp/helix/working/C24F0A5F/w/B0D20981/e
  Discovering: System.Numerics.Vectors.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Numerics.Vectors.Tests (found 1269 test cases)
  Starting:    System.Numerics.Vectors.Tests (parallel test collections = on, max threads = 8)
./RunTests.sh: line 162: 93142 Segmentation fault: 11  (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Numerics.Vectors.Tests.runtimeconfig.json --depsfile System.Numerics.Vectors.Tests.deps.json xunit.console.dll System.Numerics.Vectors.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/B0D20981/e
----- end Wed Mar 3 17:36:51 PST 2021 ----- exit code 139 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:36:12 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.Linq.Expressions.Tests.runtimeconfig.json --depsfile System.Linq.Expressions.Tests.deps.json xunit.console.dll System.Linq.Expressions.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/ADA10930/e /private/tmp/helix/working/C24F0A5F/w/ADA10930/e
  Discovering: System.Linq.Expressions.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Linq.Expressions.Tests (found 5647 of 5657 test cases)
  Starting:    System.Linq.Expressions.Tests (parallel test collections = on, max threads = 8)
./RunTests.sh: line 162: 84534 Bus error: 10           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Linq.Expressions.Tests.runtimeconfig.json --depsfile System.Linq.Expressions.Tests.deps.json xunit.console.dll System.Linq.Expressions.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/ADA10930/e
----- end Wed Mar 3 17:36:15 PST 2021 ----- exit code 138 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:36:07 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig System.IO.FileSystem.Watcher.Tests.runtimeconfig.json --depsfile System.IO.FileSystem.Watcher.Tests.deps.json xunit.console.dll System.IO.FileSystem.Watcher.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/ACC709D3/e /private/tmp/helix/working/C24F0A5F/w/ACC709D3/e
  Discovering: System.IO.FileSystem.Watcher.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.IO.FileSystem.Watcher.Tests (found 168 of 183 test cases)
  Starting:    System.IO.FileSystem.Watcher.Tests (parallel test collections = on, max threads = 8)
Unhandled exception. System.ObjectDisposedException: Safe handle has been closed.
Object name: 'SafeHandle'.
   at System.Runtime.InteropServices.SafeHandle.DangerousAddRef(Boolean& success) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs:line 148
   at System.StubHelpers.StubHelpers.SafeHandleAddRef(SafeHandle pHandle, Boolean& success) in /_/src/coreclr/System.Private.CoreLib/src/System/StubHelpers.cs:line 1244
   at Interop.EventStream.FSEventStreamUnscheduleFromRunLoop(SafeEventStreamHandle streamRef, IntPtr runLoop, SafeCreateHandle runLoopMode)
   at System.IO.FileSystemWatcher.RunningInstance.StaticWatcherRunLoopManager.UnscheduleFromRunLoop(SafeEventStreamHandle eventStream) in /_/src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs:line 224
   at System.IO.FileSystemWatcher.RunningInstance.CleanupEventStream() in /_/src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs:line 280
   at System.IO.FileSystemWatcher.RunningInstance.FileSystemEventCallback(IntPtr streamRef, IntPtr clientCallBackInfo, IntPtr numEvents, Byte** eventPaths, FSEventStreamEventFlags* eventFlags, UInt64* eventIds) in /_/src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs:line 356
./RunTests.sh: line 162: 21965 Abort trap: 6           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.IO.FileSystem.Watcher.Tests.runtimeconfig.json --depsfile System.IO.FileSystem.Watcher.Tests.deps.json xunit.console.dll System.IO.FileSystem.Watcher.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/ACC709D3/e
----- end Wed Mar 3 17:36:18 PST 2021 ----- exit code 134 ----------------------------------------------------------
```
```console
----- start Wed Mar 3 17:35:41 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/C24F0A5F/p/dotnet exec --runtimeconfig Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.runtimeconfig.json --depsfile Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.deps.json xunit.console.dll Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/C24F0A5F/w/AFBF0977/e /private/tmp/helix/working/C24F0A5F/w/AFBF0977/e
  Discovering: Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests (found 448 test cases)
  Starting:    Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests (parallel test collections = on, max threads = 8)
./RunTests.sh: line 162: 88270 Bus error: 10           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.runtimeconfig.json --depsfile Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.deps.json xunit.console.dll Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/C24F0A5F/w/AFBF0977/e
----- end Wed Mar 3 17:35:44 PST 2021 ----- exit code 138 ----------------------------------------------------------
```
```console
----- start Mon Mar 8 12:54:16 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/97BF0856/p/dotnet exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json --depsfile System.Text.Json.Tests.deps.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/97BF0856/w/A9D00955/e /private/tmp/helix/working/97BF0856/w/A9D00955/e
  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 2318 of 2325 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = on, max threads = 8)
./RunTests.sh: line 162: 54194 Bus error: 10           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.Text.Json.Tests.runtimeconfig.json --depsfile System.Text.Json.Tests.deps.json xunit.console.dll System.Text.Json.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/97BF0856/w/A9D00955/e
----- end Mon Mar 8 12:54:42 PST 2021 ----- exit code 138 ----------------------------------------------------------
```
https://dev.azure.com/dnceng/public/_build/results?buildId=1031518
```console
----- start Wed Mar 10 10:30:07 PST 2021 =============== To repro directly: =====================================================
pushd .
/tmp/helix/working/96B00891/p/dotnet exec --runtimeconfig System.IO.Compression.Tests.runtimeconfig.json --depsfile System.IO.Compression.Tests.deps.json xunit.console.dll System.IO.Compression.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing 
popd
===========================================================================================================
/private/tmp/helix/working/96B00891/w/AC6F08ED/e /private/tmp/helix/working/96B00891/w/AC6F08ED/e
  Discovering: System.IO.Compression.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.IO.Compression.Tests (found 240 test cases)
  Starting:    System.IO.Compression.Tests (parallel test collections = on, max threads = 8)
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 286
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 183
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 207
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 324
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext() in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 302
   at System.IO.Tests.StreamConformanceTests.JumpToThreadPoolAwaiter.<>c__DisplayClass4_0.<UnsafeOnCompleted>b__0(Object _) in /_/src/libraries/Common/tests/StreamConformanceTests/System/IO/StreamConformanceTests.cs:line 607
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs:line 956
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs:line 708
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:line 57
   at System.Threading.Thread.StartCallback() in /_/src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 105
./RunTests.sh: line 162: 35119 Abort trap: 6           (core dumped) "$RUNTIME_PATH/dotnet" exec --runtimeconfig System.IO.Compression.Tests.runtimeconfig.json --depsfile System.IO.Compression.Tests.deps.json xunit.console.dll System.IO.Compression.Tests.dll -xml testResults.xml -nologo -nocolor -notrait category=IgnoreForCI -notrait category=failing $RSP_FILE
/private/tmp/helix/working/96B00891/w/AC6F08ED/e
----- end Wed Mar 10 10:30:10 PST 2021 ----- exit code 134 ----------------------------------------------------------
```
These are probably GC issues....

### PR DESCRIPTION
**Pull request by sdmaclea**
These issues were likely fixed in CI by dotnet/core-eng#12776

Fixes #49110

### CODE REVIEW

File: src/libraries/tests.proj [hunk 1]
```diff
@@ -299,56 +299,6 @@
     <!-- ActiveIssue Apple Silicon No usable version of libssl was found https://github.com/dotnet/runtime/issues/49083 -->
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Security.Cryptography.Algorithms/tests/System.Security.Cryptography.Algorithms.Tests.csproj" />
     <ProjectExclusions Include="$(MSBuildThisFileDirectory)System.Security.Cryptography.OpenSsl/tests/System.Security.Cryptography.OpenSsl.Tests.csproj" />
-    <!-- ActiveIssue XUnit crashes on Apple Silicon https://github.com/dotnet/runtime/issues/49110 -->
```
Comment (safern) [inline src/libraries/tests.proj:302]: Could all these that were crashing uncover other test failures? Let's make sure before merging that the build is still green.

**Review by safern — APPROVED** [review]

**Comment by sdmaclea** [comment]
Manually triggered https://dev.azure.com/dnceng/public/_build/results?buildId=1119668&view=results

**Comment by ericstj** [comment]
@sdmaclea does this need a rerun or do you think these weren't fixed by  dotnet/core-eng#12776?

**Comment by sdmaclea** [comment]
These weren't fixed by dotnet/core-eng#12776.  

Unfortunately most of these issues do not reproduce (easily?) locally anymore.  

The original runs didn't generate core dumps.  The coredumps have since been fixed.  I planned to rebase and rerun these to see if we could get some more info.

/cc @janvorli @mangod9 @sandreenko

**Comment by sdmaclea** [comment]
Rebased and manually triggered https://dev.azure.com/dnceng/public/_build/results?buildId=1176673&view=results

**Comment by sdmaclea** [comment]
There are some coredumps for a bus error and a segmentation fault https://dev.azure.com/dnceng/public/_build/results?buildId=1176673

There is also this interesting delegate test failure
```console
  Starting:    System.Runtime.Tests (parallel test collections = on, max threads = 8)
    System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithSomeExplicitValues [FAIL]
      Assert.Equal() Failure
                                       â†“ (pos 38)
      Expected: Â·Â·Â· -101, -3, 104, -5, 106, -7, 108, 9.1, 111.12
      Actual:   Â·Â·Â· -101, -3, 104, -5, 4294967291, -7, 108, 9.1, 111.12
                                       â†‘ (pos 38)
      Stack Trace:
        /_/src/libraries/System.Runtime/tests/System/DelegateTests.cs(253,0): at System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithSomeExplicitValues()
    System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithMissingValues [FAIL]
      Assert.Equal() Failure
                                       â†“ (pos 33)
      Expected: Â·Â·Â·, 2, -1, -3, 4, -5, 6, -7, 8, 9.1, 11.12
      Actual:   Â·Â·Â·, 2, -1, -3, 4, -5, 4294967291, -7, 8, 9.1, 11.12
                                       â†‘ (pos 33)
    System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithAllExplicitValues [FAIL]
      Assert.Equal() Failure
                                       â†“ (pos 45)
      Expected: Â·Â·Â·1, -103, 104, -105, 106, -107, 108, 109.1, 111.12
      Actual:   Â·Â·Â·1, -103, 104, -105, 4294967191, -107, 108, 109.1, 111.12
                                       â†‘ (pos 45)
      Stack Trace:
        /_/src/libraries/System.Runtime/tests/System/DelegateTests.cs(228,0): at System.Tests.DelegateTests.DynamicInvoke_DefaultParameter_AllPrimitiveParametersWithAllExplicitValues()
```

Looks like an ABI issue which was covered by the disabled tests.  Perhaps it was fixed by #53250 which was merged after this was triggered.

**Comment by sdmaclea** [comment]
This run the same test suites failed. https://dev.azure.com/dnceng/public/_build/results?buildId=1177129

But the interesting case is gone.  Perhaps it is fixed or intermittent.

**Pull request merged**